physical/cassandra.go

```diff
@@ -72,6 +75,39 @@ func newCassandraBackend(conf map[string]string, logger log.Logger) (Backend, er
 	connectStart := time.Now()
 	cluster := gocql.NewCluster(hosts...)
 	cluster.Keyspace = keyspace
+
+	cluster.ProtoVersion = 2
+	if protoVersionStr, ok := conf["protocol_version"]; ok {
+		protoVersion, err := strconv.Atoi(protoVersionStr)
+		if err != nil {
+			return nil, fmt.Errorf("'protocol_version' must be an integer")
+		}
+		cluster.ProtoVersion = protoVersion
+	}
+
+	if username, ok := conf["username"]; ok {
+		if cluster.ProtoVersion < 2 {
+			return nil, fmt.Errorf("Authentication is not supported with protocol version < 2")
+		}
+		authenticator := gocql.PasswordAuthenticator{Username: username}
+		if password, ok := conf["password"]; ok {
+			authenticator.Password = password
+		}
+		cluster.Authenticator = authenticator
+	}
+
+	if connTimeoutStr, ok := conf["connection_timeout"]; ok {
+		connectionTimeout, err := strconv.Atoi(connTimeoutStr)
+		if err != nil {
+			return nil, fmt.Errorf("'connection_timeout' must be an integer")
+		}
+		cluster.Timeout = time.Duration(connectionTimeout) * time.Millisecond
```

Comment:
Just noticed that in [this file](https://github.com/hashicorp/vault/blob/d06b36ce8d68a3366eb00c95a2f731891d52af79/builtin/logical/cassandra/util.go#L36), this parameter is specified in seconds rather than milliseconds. Should we align the two for consistency?